src/components/dependencies/RescoringModal.js

```diff
@@ -2179,34 +2325,61 @@ const RescoringModal = ({
         },
       ]
     })).values()]: []))
-  }, [rescorings, setSprints])
-
-  React.useEffect(() => {
-    if (!rescorings) return
+    setSprintsLoading(false)
+  }, [rescorings, setSprints, rescoringsLoading])
 
-    if (selectedSprints.length === 0) {
-      setFilteredRescorings(rescorings)
-      return
-    }
+  const allowedRescorings = filteredRescorings?.filter((rescoring) => {
+    return (
+      selectedRescorings.find((r) => rescoringIdentity(r) === rescoringIdentity(rescoring))
+      && rescoring.severity !== rescoringProposalSeverity(rescoring)
+    )
+  })
+  const filteredOutRescoringsLength = filteredRescorings ? selectedRescorings.length - allowedRescorings.length : 0
 
-    setFilteredRescorings(rescorings.filter((rescoring) => {
-      const sprintName = sprintNameForRescoring(rescoring)
-      return selectedSprints.some((sprint) => sprintName === sprint)
+  React.useEffect(() => {
+    setFilteredRescorings(rescorings.filter(rescoring => {
+      for (const filter of Object.values(filters)) {
```

Comment:
Rather use `Object.values(filters).reduce()` instead of `for` loop

Reply:
Are there any benefits else than personal preference?
This implementation would early-exit and spare iterations, whereas `reduce` always completes its iterations 🤔

Reply:
I would have argued with consistency in our code base. However, I just checked and figured we also have this in place in 2 other occurrences, so I'd leave it up to you to choose what fits here best :-)